xstream/nodes/source_node.go

```diff
@@ -190,18 +192,25 @@ func (m *SourceNode) getConf(ctx api.StreamContext) map[string]interface{} {
 	conf, err := common.LoadConf(confPath)
 	props := make(map[string]interface{})
 	if err == nil {
-		cfg := make(map[string]map[string]interface{})
+		cfg := make(map[interface{}]interface{})
```

Comment:
The original design just return the map[interface{}]interface{} prop value to the source. And the source config can do the conversion there

The root cfg seems always have a string key "default", do we need to change the map key type here?

Reply:
I changed the definition is that it has a bug when trying to load with hierarchy configurations. As in below, the `ClientId` under `optional` cannot be loaded.

```yaml
default:
  protocol: tcp
  server: localhost
  type: mqtt
  optional:
    ClientId: client1
    Username: user1
```